packages/openssl/build.sh

```diff
@@ -17,21 +17,24 @@ TERMUX_PKG_REPLACES="openssl-tool (<< 1.1.1b-1), openssl-dev"
 termux_step_configure() {
 	# Certain packages are not safe to build on device because their
 	# build.sh script deletes specific files in $TERMUX_PREFIX.
-	if $TERMUX_ON_DEVICE_BUILD; then
+	if [[ "$TERMUX_ON_DEVICE_BUILD" == 'true' ]]; then
 		termux_error_exit "Package '$TERMUX_PKG_NAME' is not safe for on-device builds."
 	fi
 
 	CFLAGS+=" -DNO_SYSLOG"
 
-	perl -p -i -e "s@TERMUX_CFLAGS@$CFLAGS@g" Configure
-	rm -Rf $TERMUX_PREFIX/lib/libcrypto.* $TERMUX_PREFIX/lib/libssl.*
-	test $TERMUX_ARCH = "arm" && TERMUX_OPENSSL_PLATFORM="android-arm"
-	test $TERMUX_ARCH = "aarch64" && TERMUX_OPENSSL_PLATFORM="android-arm64"
-	test $TERMUX_ARCH = "i686" && TERMUX_OPENSSL_PLATFORM="android-x86"
-	test $TERMUX_ARCH = "x86_64" && TERMUX_OPENSSL_PLATFORM="android-x86_64"
-	./Configure $TERMUX_OPENSSL_PLATFORM \
-		--prefix=$TERMUX_PREFIX \
-		--openssldir=$TERMUX_PREFIX/etc/tls \
+	sed -i "s@TERMUX_CFLAGS@$CFLAGS@g" Configure
+	rm -rf "$TERMUX_PREFIX/lib"/libcrypto.* "$TERMUX_PREFIX/lib"/libssl.*
+
+	local TERMUX_OPENSSL_PLATFORM="android-${TERMUX_ARCH}"
+	case "$TERMUX_ARCH" in
+		"aarch64") TERMUX_OPENSSL_PLATFORM="android-arm64";;
+		"i686") TERMUX_OPENSSL_PLATFORM="android-x86";;
+	esac
```

Comment:
Maybe add `arm|x86_64)` for direct, and `*)` to error out so that new arches can be explicitly added with correct mapping, like `riscv`.

Reply:
Good point.